June 2021 news

aeslaughter opened the pull request:
@idaholab/moose-ccb  **Please add your work from June 2021 to the 2021_06.md file.** 

This is different to what we have done in the past, but the old way always caused confusion. So, if you did the work in June, but it in June. The most important thing is just getting items mentioned, so don't worry too much about the date.

You can also add items that you may have merged this month to the July file (2021_07.md)

(refs #11447)

aeslaughter committed:
Jun/July 2021 news

(refs #11447)

dschwen commented:
I'll add a bit about multiple inputs

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/772462/) on 7c60816 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/18275/site)

This comment will be updated on new commits.

dschwen commented on modules/doc/content/newsletter/2021_06.md:
```diff
@@ -2,3 +2,5 @@
 
 
```
## Multiple input file support

Users can now supply multiple input file names after the `-i` option. The supplied inputs are successively merged into one simulation specification, with later files adding and overwriting parameters. This allows users to factor out common parameters (e.g. solver options, postprocessors, outputs) from a set of similar input files

 ```
 ./mooseapp-opt -i common_parameters.i simulation_1.i 
```

The output file names will be based off the _last_ input file name specified (unless the `Outputs/file_base` parameter is specified). When multiple inputs are specified, MOOSE will inform the user about parameter overrides at the top of the simulation output. 

To assist users with refactoring input files several new functions we added to the `hit` commandline tool.

- `hit merge -output outfile.i infile1.i infile2.i ...` will produce a single file `outfile.i` that is a merge of the `infile*.i` inputs (which would result in the same MOOSE simulation as the multiple input files).
- `hit diff left.i right.i` or `hit diff -left left1.i left2.i ... -right right1.i right2.i ...` performs a diff on the left and right files (in the second example merging all left and right files respectively first). This diff is not sensitive to order, formatting, or comments in the input files.
- `hit common file1.i file2.i ... > common.i` will extract all parameters that are common to the specified input files (and have the same values). This can be used as the first step in factoring out common settings into a single input file.
- `hit subtract common.i simulation_1_full.i > simulation_1.i` removes the parameters in `common.i` from `simulation_1_full.i`, creating `simulation_1.i`. This is the second step in factoring out common parameters from a set of input files. The resulting file can be run as `./mooseapp-opt -i common.i simulation_1.i` and will result in teh same simulation as `./mooseapp-opt -i simulation_1_full.i`.

dschwen replied:
am I supposed to make a PR to your repo @aeslaughter ?

aeslaughter replied:
members of the CCB can just push directly to my branch, non members need to make a PR to my repo

dschwen replied:
Well, I'm a CCB member, but I get

```
schwd$ git push --set-upstream aeslaughter  mooseletter-11447
ERROR: Authentication error: Authentication required: You must have push access to verify locks
error: failed to push some refs to 'github.com:aeslaughter/moose'
```

dschwen replied:
You can cherry-pick c76c01d06f8d8c9296c139bbdc5e6a35e5921a40

aeslaughter replied:
You are doing something wrong. I usually do something like `git push aeslaughter HEAD:mooseletter-11447` 

![Screen Shot 2021-07-12 at 8 48 03 AM](https://user-images.githubusercontent.com/341333/125308394-1b1e5a80-e2ee-11eb-988d-7cb779fb4fdf.png)

dschwen replied:
```
(moose) fn606041:moose schwd$ git push aeslaughter HEAD:mooseletter-11447
ERROR: Authentication error: Authentication required: You must have push access to verify locks
error: failed to push some refs to 'github.com:aeslaughter/moose'
```

in my `.git/config` I have the proper ssh git url:
```
(moose) fn606041:moose schwd$ git push aeslaughter HEAD:mooseletter-11447
ERROR: Authentication error: Authentication required: You must have push access to verify locks
error: failed to push some refs to 'github.com:aeslaughter/moose'
```

aeslaughter replied:
OK, I will grab your commit.

lindsayad replied:
@dschwen I was able to push to @aeslaughter 's branch this morning

dschwen committed:
Add multiple input section (#11447)

lindsayad committed:
Alex updates

aeslaughter committed:
Aug/Sept news

aeslaughter commented:
Please add your contributions for Aug. to this PR and we can finally get this merged.

@crswong888 Can you add a few sentences about your MooseDocs changes?

crswong888 committed:
MooseDocs multiconfigs news

(refs #11447 #18700)

cticenhour commented on modules/doc/content/newsletter/2021_09.md:
```diff
@@ -5,3 +5,40 @@
   solve. This allows the user to check if the solve converged, or if it did
   converged, by what criteria.
+
+## Compressible Navier-Stokes Finite Volume (CNSFV)
+
+Through the `navier_stokes` module, MOOSE now has the ability to model transonic
+and supersonic flows using either Harten, Lax, van Leer, Contact (HLLC) or
+Kurganov-Tadmor discretization strategies for the advective flux. The
+compressible Euler equations are implemented using the finite volume method. We
+have also implemented a variety of slope limiters, including van Leer, min-mod,
+first and second order upwind, central differencing, and QUICK, that can be used
+when interpolating quantities from cell centroids to faces. We do not recommend
+this compressible fluid flow implementation for low Mach number flows as results
+can be overly diffusive when the solution is not smooth, e.g. in high Rayleigh
+number natural convection simulations.
+
+### libMesh update
+
+We updated libMesh. Here is a summary of the changes:
+
+  - Mimic stl and boost constructors in our allocators
+  - Try changing max order for L2 families
+  - Add LumpedMassMatrix class
+  - Test custom FE reinit
+  - `$LIBMESH_BENCHMARK` selection of benchmarking examples
+  - RB: skip reinit when there are no DOFs on Elem
+  - Command-line override of default Metis-vs-Parmetis
+  - init_names() at more points
+  - Fix FE side unit tests with --enable-complex
+  - Begin contains_vertex_of with node equality tests
+  - chunked_mapvector storage option
+  - Weaken partitioner_test for SFC on 8+ processors
+  - Bugfix for PR libmesh/libmesh#2940
+  - Revert "interior_parent fix on distributed refined multi-elem_dim meshes"
+  - interior_parent fix on distributed refined multi-elem_dim meshes
+  - RB EIM update
+  - Fix for deprecated warning in SLEPc not-quite-3.15
+  - Use pool allocator in mapvector
+  - Check for gdb in command line in print_trace when compiling with no gdb backtrace by default
```
This is a repeat of entries in July. Which month is this applicable to?

aeslaughter replied:
I will fix it.

aeslaughter committed:
Fix duplicate in failed copy/rebase

(refs #11447)

aeslaughter commented:
Just realized that I forgot to pull down my branch first, so if anyone added something after my commit I deleted it. Rookie mistake by me.

dschwen commented:
Should all still be in the reflog or did it expire?

On Wed, Sep 15, 2021, 4:47 PM Andrew E Slaughter ***@***.***>
wrote:

> Just realized that I forgot to pull down my branch first, so if anyone
> added something after my commit I deleted it. Rookie mistake by me.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/18275#issuecomment-920441985>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABRMPRDVBLTSOTDMJPO3TTUCEPANANCNFSM4763ODWA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

aeslaughter committed:
Merge pull request #34 from crswong888/mooseletter-11447

MooseDocs multiconfigs news

moosebuild commented:
Job [Generate and verify coverage](https://civet.inl.gov/job/821180/) on a42abbb wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>4fa2e0</th>
      <th colspan=4>#18275 a42abb</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>81.51%</td>
      <td>81.51%</td>
      <td>-</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>69411</td>
      <td>69411</td>
      <td>-</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15749</td>
      <td>15749</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/18275/coverage/framework.html)

## Modules coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>4fa2e0</th>
      <th colspan=4>#18275 a42abb</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>86.99%</td>
      <td>86.99%</td>
      <td>-</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>90751</td>
      <td>90751</td>
      <td>-</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>13567</td>
      <td>13567</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/18275/coverage/modules.html)

## Full coverage reports

<details><summary>Reports</summary>

- [framework](https://mooseframework.inl.gov/docs/PRs/18275/coverage/framework)
- [chemical_reactions](https://mooseframework.inl.gov/docs/PRs/18275/coverage/chemical_reactions)
- [combined](https://mooseframework.inl.gov/docs/PRs/18275/coverage/combined)
- [contact](https://mooseframework.inl.gov/docs/PRs/18275/coverage/contact)
- [external_petsc_solver](https://mooseframework.inl.gov/docs/PRs/18275/coverage/external_petsc_solver)
- [fluid_properties](https://mooseframework.inl.gov/docs/PRs/18275/coverage/fluid_properties)
- [fsi](https://mooseframework.inl.gov/docs/PRs/18275/coverage/fsi)
- [functional_expansion_tools](https://mooseframework.inl.gov/docs/PRs/18275/coverage/functional_expansion_tools)
- [geochemistry](https://mooseframework.inl.gov/docs/PRs/18275/coverage/geochemistry)
- [heat_conduction](https://mooseframework.inl.gov/docs/PRs/18275/coverage/heat_conduction)
- [level_set](https://mooseframework.inl.gov/docs/PRs/18275/coverage/level_set)
- [misc](https://mooseframework.inl.gov/docs/PRs/18275/coverage/misc)
- [navier_stokes](https://mooseframework.inl.gov/docs/PRs/18275/coverage/navier_stokes)
- [peridynamics](https://mooseframework.inl.gov/docs/PRs/18275/coverage/peridynamics)
- [phase_field](https://mooseframework.inl.gov/docs/PRs/18275/coverage/phase_field)
- [porous_flow](https://mooseframework.inl.gov/docs/PRs/18275/coverage/porous_flow)
- [ray_tracing](https://mooseframework.inl.gov/docs/PRs/18275/coverage/ray_tracing)
- [rdg](https://mooseframework.inl.gov/docs/PRs/18275/coverage/rdg)
- [richards](https://mooseframework.inl.gov/docs/PRs/18275/coverage/richards)
- [stochastic_tools](https://mooseframework.inl.gov/docs/PRs/18275/coverage/stochastic_tools)
- [tensor_mechanics](https://mooseframework.inl.gov/docs/PRs/18275/coverage/tensor_mechanics)
- [xfem](https://mooseframework.inl.gov/docs/PRs/18275/coverage/xfem)

</details>


This comment will be updated on new commits.

GiudGiud committed:
Add my june july and august contributions, refs #11447

GiudGiud committed:
Add Josh s UO PR to newsletter in august

roystgnr committed:
Add highlights of the June libMesh update

roystgnr committed:
September libMesh updates in newsletter

Just in case I let this slip my mind again later, at least the big
update will be in already.

fdkong committed:
Fande's contributions

cticenhour commented on modules/doc/content/newsletter/2021_07.md:
```diff
@@ -0,0 +1,42 @@
+
+## Bug Fixes and Minor Enhancements
+
+- Minor fixes to links in the documentation, thanks to feedback in #18425
```
```suggestion
- Minor fixes to links in the documentation, thanks to feedback in [#18425](https://github.com/idaholab/moose/discussions/18425)
```

cticenhour commented on modules/doc/content/newsletter/2021_07.md:
```diff
@@ -0,0 +1,42 @@
+
+- Minor fixes to links in the documentation, thanks to feedback in #18425
+
+### libMesh update
+
+We updated libMesh. Here is a summary of the changes:
```
```suggestion
## libMesh-level changes
```

Slightly more consistency

cticenhour commented on modules/doc/content/newsletter/2021_08.md:
```diff
@@ -15,6 +15,34 @@ method `computeQpResidual()`. As a caution, if using user objects to compute
+
+## Improvements to Navier Stokes Flow capabilities
+
+- Thermal boundary conditions, including radiative and convective heat transfer boundary conditions
+  were reworked to support primitive, mixed and conservative variable sets for fluid flow.
```
```suggestion
- Thermal boundary conditions, including radiative and convective heat transfer boundary conditions,
  were reworked to support primitive, mixed, and conservative variable sets for fluid flow.
```

cticenhour commented on modules/doc/content/newsletter/2021_08.md:
```diff
@@ -25,3 +53,35 @@ ensure the derivatives in the `ADReal` value are populated.
+- When a block or boundary is mispelled, the code will now output the name of the
+  block or boundary in addition to a newly generated id, which will simplify the
+  search for the issue.
+- Dependendy resolution between user objects and auxiliary kernels will now be cognizant
```
```suggestion
- Dependency resolution between user objects and auxiliary kernels will now be cognizant
```

cticenhour commented on modules/doc/content/newsletter/2021_08.md:
```diff
@@ -25,3 +53,35 @@ ensure the derivatives in the `ADReal` value are populated.
+  search for the issue.
+- Dependendy resolution between user objects and auxiliary kernels will now be cognizant
+  of their respective execution flags.
+- If the dependency resolution between UOs and auxiliary kernel is known to fail in a given case,
```
```suggestion
- If the dependency resolution between user objects and auxiliary kernels is known to fail in a given case,
```

cticenhour reviewed (changes requested):
A few more suggested changes. Waiting on a final entry regarding live print from @friedmud

friedmud committed:
Add Information about Perf Graph Live

GiudGiud committed:
Apply suggestions from code review

lgtm thanks for fixing this

Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>
